### PR TITLE
rust plugin: add features for conditional compilation

### PR DESCRIPTION
**Pull request by cholcombe973**
This adds support for optional dependencies that can be built into the binary.

### CODE REVIEW

**Comment by cholcombe973** [comment]
@sergiusens I'm not quite sure how to change my code around to make use of `get_build_properties`

**Comment by sergiusens** [comment]
hey @cholcombe973 I made a pull request with no unit tests to get you going.

**Comment by cholcombe973** [comment]
@sergiusens which PR is it?

**Comment by sergiusens** [comment]
https://github.com/cholcombe973/snapcraft/pull/1

**Comment by sergiusens** [comment]
It would be nice to add a unit test here if possible.

File: snapcraft/tests/test_plugin_rust.py [hunk 1]
```diff
@@ -64,15 +63,19 @@ def test_schema(self):
     def test_build(self, run_mock):
         plugin = rust.RustPlugin('test-part', self.options,
                                  self.project_options)
+        plugin.options.rust_features = ['conditional-compilation']
```
Comment (elopio) [inline snapcraft/tests/test_plugin_rust.py]: I would prefer this to be in a separate test, to keep each unit test small and testing a single thing.

Comment (cholcombe973) [inline snapcraft/tests/test_plugin_rust.py]: That's fine.  I can add another test

**Comment by elopio** [comment]
@cholcombe973 thanks again for the contribution.
I've finished the tests in https://github.com/snapcore/snapcraft/pull/1029

**Pull request closed without merging**